articles/4_flex.md

```diff
@@ -0,0 +1,141 @@
+
+[Codepen Demo](http://codepen.io/raemadeline/pen/jrreaQ)
+
+#### `flex-flow`
+
+This is just a shorthand to combine `flex-direction` and `flex-wrap`. The syntax is:
+
+```css
+.parent {
+  flex-flow: <'flex-direction'> || <'flex-wrap'>
+}
+```
+
+#### `justify-content`
+
+This determines the spacing of child elements along the main axis. This one is easier to understand visually, so here's an example:
+
+![Justify Content](../images/justify_content.png)
+
+To play around with these values, here's the [Codepen Demo](http://codepen.io/raemadeline/pen/BzzqxV). Try changing around other properties like `flex-direction` to see how these change.
+
+#### `align-items`
+
+`align-items` determines the spacing of child elements along the *secondary* axis, the one perpendicular to the axis specified in `flex-direction`.
+
+http://codepen.io/raemadeline/pen/WxGeNd
+
+#### `align-content`
+
+`align-content` is the same as `align-items`, but it handles the full range of child elements as a block instead of arranging each element in their line. For a better explanation, see the visual below (thanks css-tricks.com)
+
+![align-content](../images/align-content.png)
+
+### The Child Elements
+
+By applying the above properties to the parent element, you can create a lot of awesome layouts. However, you may occasionally want to customize a specific child element of the parent separately from the rest. These are properties you can add to the direct children of a flex parent.
+
+#### `order`
+
+This overrides the ordering of the elements as determined by the parent. By default, every child's `order` is determined by the layout. Generally, you should avoid this and just arrange your HTML such that the elements start in the correct order, but this is useful with responsive designs. See this [Codepen Demo](http://codepen.io/raemadeline/pen/PzGwGw) and try resizing the window. If you look at the HTML, the sidebar is before the main content, but its `order` gets overridden at smaller screen sizes to account for users looking at the page on mobile.
+
+#### `flex-grow`
+
+By default, assuming all `flex` children have the same size internally, they will all grow an equal amount to fill the remaining space left in the parent (assuming their `width` and `height` have not yet been fixed). This property determines the priority of the child to fill up more space. All children have a default of `flex-grow: 1;`, so if a child is given `flex-grow: 2`, it will take up twice as much remaining space as its siblings.
+
+![flex-grow](../images/flex-grow.png)
+
+#### `flex-shrink`
+
+`flex-shrink` is the complement of `flex-grow`. In the case when the parent is too small to comfortably fit all the children, any child with a `flex-shrink` property that's greater than zero will shrink to allow all of the other children to fit. By default, all `flex` children have `flex-shrink: 0`, so this will not happen unless you explicitly declare it.
```

Comment:
Default is actually 1. Similar to my comment on flex-grow, doesn't mean they would always automatically shrink, but will do it sometimes unpredictably if you don't tell it not to (as I've unfortunately discovered)